qiskit/qpy/binary_io/value.py

```diff
@@ -260,17 +289,36 @@ def _read_parameter_expression(file_obj):
 
 
 def _read_parameter_expression_v3(file_obj, vectors):
+
     data = formats.PARAMETER_EXPR(
         *struct.unpack(formats.PARAMETER_EXPR_PACK, file_obj.read(formats.PARAMETER_EXPR_SIZE))
     )
+
     from sympy.parsing.sympy_parser import parse_expr
 
+    payload = file_obj.read(data.expr_size)
+
     if _optional.HAS_SYMENGINE:
-        import symengine
+        from symengine import sympify
+        from symengine.lib.symengine_wrapper import load_basic
+
+        try:
+            expr_ = load_basic(payload)
 
-        expr_ = symengine.sympify(parse_expr(file_obj.read(data.expr_size).decode(common.ENCODE)))
+        except RuntimeError:
+            warnings.warn("Symengine loading failed, trying with sympy...")
+            expr_ = sympify(parse_expr(payload.decode(common.ENCODE)))
     else:
-        expr_ = parse_expr(file_obj.read(data.expr_size).decode(common.ENCODE))
+        warnings.warn("Symengine not found, trying with sympy...")
+
+        try:
+            expr_ = parse_expr(payload.decode(common.ENCODE))
+
+        except UnicodeDecodeError as exc:
```

Comment:
This is the error type raised when you serialize with symengine and try to deserialize with sympy. Should I make this catch a more generic exception??